src/components/sidebar/SidebarMenu.tsx

```diff
@@ -0,0 +1,53 @@
+import { SidebarItem } from "./SidebarItem";
+import { CloseIcon } from "../icons";
+import { MouseEventHandler } from "react";
+import { MenuItem } from "../../types";
+import { cn } from "../../utils/cn";
+
+type SidebarMenuProps = {
+  menuItems: MenuItem[];
+  handleOpenSidebar: MouseEventHandler<HTMLButtonElement>;
+  handleOpenSubMenu: (menuItem: MenuItem) => void;
+  showCloseButton: boolean;
+  expandedMenuItem?: MenuItem | null;
+  isSubMenu?: boolean;
+};
+
+export const SidebarMenu = ({
+  handleOpenSidebar,
+  handleOpenSubMenu,
+  menuItems,
+  showCloseButton,
+  expandedMenuItem,
+  isSubMenu,
+}: SidebarMenuProps) => {
+  return (
+    <div
+      className={cn(
+        "flex flex-col h-full w-3/5 md:w-2/6 lg:w-3/12 bg-primary-50 shadow-lg py-14 px-12 gap-9 overflow-auto z-50",
+        {
+          "z-40": isSubMenu,
+        }
+      )}
+    >
+      <button onClick={handleOpenSidebar}>
+        <CloseIcon
+          className={cn("size-6", { invisible: !showCloseButton })}
```

Comment:
i would suggest going for the positive representation: { visible: showCloseButton } as it is more intuitive